test/serializers/default_include_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class DefaultIncludeTest < ActiveSupport::TestCase
+      class Blog < ActiveModelSerializers::Model
+        attr_accessor :id, :name, :posts
+      end
+      class Post < ActiveModelSerializers::Model
+        attr_accessor :id, :title, :author
+      end
+      class Author < ActiveModelSerializers::Model
+        attr_accessor :id, :name
+      end
+      class BlogSerializer < ActiveModel::Serializer
+        attributes :id
+        attribute :name, key: :title
+
+        has_many :posts
+      end
+      class PostSerializer < ActiveModel::Serializer
+        default_include [:author]
+        attributes :id, :title
+
+        has_one :author
```

Comment:
Quoting from my comment in https://github.com/rails-api/active_model_serializers/issues/1843#issuecomment-232387846

> - serializers have an `attributes` methods. It executes `read_attribute_for_serialization` on each attribute in `self.class._attributes`.
  - the list of serialized attributes can be limited by passing in an argument `options[:fields]`
- serializers have a list of defined `associations`
  - for each associated, a serializer is found, and we call `attributes` on it.
    - we pass in to attributes `options[:includes][association_name]`, see serializer for what this does, like fields

So, that 

| | render options | serializer attributes under JSON adapter | serializer attributes under JSONAPI adapter |
|---|------------- |--------------- | ------------------|
| `has_one :author` | none | `AuthorSerializer.new(object.author).attributes` | not serialized
| `has_one :author` | `include: [:author]` | `AuthorSerializer.new(object.author).attributes` | `AuthorSerializer.new(object.author).attributes`
| `has_one :author` | `include: [:author], fields: { authors: [:name]}` | `AuthorSerializer.new(object.author).attributes([:name])` | `AuthorSerializer.new(object.author).attributes([:name])`
| `has_one :author, include: :all` | none | `AuthorSerializer.new(object.author).attributes` | `AuthorSerializer.new(object.author).attributes`
| `has_one :author, include: [:name]` | none | `AuthorSerializer.new(object.author).attributes([:name])` | `AuthorSerializer.new(object.author).attributes([:name])`
| `has_one :author, include: :all` | `include: [:author], fields: {authors: [:name]}` | `AuthorSerializer.new(object.author).attributes([:name])` | `AuthorSerializer.new(object.author).attributes([:name])`